src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1438,12 +1456,6 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             return;
         }
 
-        // Clear the regex pattern tree so the renderer does not try to render them while scrolling
-        // We're **NOT** taking the lock here unlike _scrollbarChangeHandler because
-        // we are already under lock (since this usually happens as a result of writing).
-        // TODO GH#9617: refine locking around pattern tree
-        _terminal->ClearPatternTree();

```

Comment:
This moved into `Terminal`. This makes the code more robust (it's also a private function now).